platforms/cuda/src/CudaKernels.cpp

```diff
@@ -2045,8 +2185,80 @@ double CudaCalcNonbondedForceKernel::execute(ContextImpl& context, bool includeF
             cuEventRecord(pmeSyncEvent, pmeStream);
             cu.restoreDefaultStream();
         }
+
+        // As written, we check only the Electrostatic grid pointer to get here.  We could separate them out, but for
+        // now we assume that LJPME can only be used if electrostatic PME is also active.
+        if(doLJPME){
+            if (usePmeStream)
+                cu.setCurrentStream(dispersionPmeStream);
```

Comment:
Let me muck around with it for a while, and then I'll send a PR to your branch.  While I'm at it, I'd like to see if I can eliminate some of the duplicated code.  `pme.cu` and `ljpme.cu` are nearly identical, so I'd like to see if I can combine them with just a few ifdefs or something similar for the varying parts.